embodiedscan/models/data_preprocessors/data_preprocessor.py

```diff
@@ -0,0 +1,584 @@
+    modality 3D detection tasks.
+
+    It provides the data pre-processing as follows
+
+    - Collate and move image and point cloud data to the target device.
+
+    - 1) For image data:
+
+      - Pad images in inputs to the maximum size of current batch with defined
+        ``pad_value``. The padding size can be divisible by a defined
+        ``pad_size_divisor``.
+      - Stack images in inputs to batch_imgs.
+      - Convert images in inputs from bgr to rgb if the shape of input is
+        (3, H, W).
+      - Normalize images in inputs with defined std and mean.
+      - Do batch augmentations during training.
+
+    - 2) For point cloud data:
+
+      - If no voxelization, directly return list of point cloud data.
+      - If voxelization is applied, voxelize point cloud according to
+        ``voxel_type`` and obtain ``voxels``.
+
+    Args:
+        voxel (bool): Whether to apply voxelization to point cloud.
+            Defaults to False.
+        voxel_type (str): Voxelization type. Two voxelization types are
+            provided: 'hard' and 'dynamic', respectively for hard voxelization
+            and dynamic voxelization. Defaults to 'hard'.
+        voxel_layer (dict or :obj:`ConfigDict`, optional): Voxelization layer
+            config. Defaults to None.
+        batch_first (bool): Whether to put the batch dimension to the first
+            dimension when getting voxel coordinates. Defaults to True.
+        max_voxels (int, optional): Maximum number of voxels in each voxel
+            grid. Defaults to None.
+        mean (Sequence[Number], optional): The pixel mean of R, G, B channels.
+            Defaults to None.
+        std (Sequence[Number], optional): The pixel standard deviation of
+            R, G, B channels. Defaults to None.
+        pad_size_divisor (int): The size of padded image should be divisible by
+            ``pad_size_divisor``. Defaults to 1.
+        pad_value (float or int): The padded pixel value. Defaults to 0.
+        pad_mask (bool): Whether to pad instance masks. Defaults to False.
+        mask_pad_value (int): The padded pixel value for instance masks.
+            Defaults to 0.
+        pad_seg (bool): Whether to pad semantic segmentation maps.
+            Defaults to False.
+        seg_pad_value (int): The padded pixel value for semantic segmentation
+            maps. Defaults to 255.
+        bgr_to_rgb (bool): Whether to convert image from BGR to RGB.
+            Defaults to False.
+        rgb_to_bgr (bool): Whether to convert image from RGB to BGR.
+            Defaults to False.
+        boxtype2tensor (bool): Whether to convert the ``BaseBoxes`` type of
+            bboxes data to ``Tensor`` type. Defaults to True.
+        non_blocking (bool): Whether to block current process when transferring
+            data to device. Defaults to False.
+        batch_augments (List[dict], optional): Batch-level augmentations.
+            Defaults to None.
```

Comment:
Update the docstring: batchwise_inputs: Pack the input as a batch of samples with 1-N frames for the continuous 3D perception setting.